### DISCUSSION TITLE
406 error on search

### DISCUSSION THREAD

**Discussion by AdrianEdelen** [crunchy-labs/crunchy-cli#432] [Q&A]
Getting a 406 on search but only when using Popen. and only about 90 percent of the time.
this worked perfectly performing searches relatively often (but not more than ~once a minute for a few hours, and now seems to be hit or miss (with about 90% miss)

```
def crunchy_cli_search(series_title, attempt_count = 0):
    output_template = '"{{series.title}},https://www.crunchyroll.com/series/{{series.id}}"'
    command = [ "crunchy-cli",
        "--credentials", crunchyroll_creds, 
        "search",
        "-o", output_template,
        f'{series_title}'
    ]

    print("Performing crunchy roll search")

    result_dict = {}


    try:
        with Popen(command, stdout=PIPE, bufsize=1, universal_newlines=True) as p:
            for line in p.stdout:
                print(line, end='')
                title, url = line.strip('"').split(',')
                result_dict[title.strip()] = url.strip()
            if p.returncode != 0:
                raise CalledProcessError(p.returncode, p.args)
    except CalledProcessError as e:
        print(f"Command '{e.cmd}' returned non-zero exit status {e.returncode}.")
    except Exception as e:
        print(F"An error occurred: {e}")

    if series_title in result_dict:
        print(f"Exact series match found, slug generated {result_dict[series_title]}")
        
        return result_dict[series_title]
    else:
        print(f"No exact match found for {series_title}, attempt: {attempt_count+1}")
        time.sleep(5)
        if attempt_count < 5:
            crunchy_cli_search(series_title=series_title, attempt_count=attempt_count+1)
        raise Exception(f"No exact match found for {series_title}")
```

        
  I am attempting to build series URLs by getting the series id and using that as a slug, which mostly works.
  
  here is the output I am getting in my terminal from crunchy-cli 
`  :: trailing characters at line 1 column 5 at 1:5 (https://www.crunchyroll.com/auth/v1/token): 406 Not Acceptable`

**Comment by bytedream** [comment]
Which version do you use?

**Reply by AdrianEdelen** [reply]
3.6.6

**Reply by AdrianEdelen** [reply]
I am probably just going to wrap the API in python at some point so I can have a little more control (and don't need to use subprocess) but for now just using the cli as a stop gap.

**Reply by AdrianEdelen** [reply]
~~For what it is worth, it seems to be working again today,~~ so I am wondering if I hit some sort of rate limiting, but a 406 is weird for that, I would assume it would be a 429 (though, i guess they don't consider it an official API so :shrug:)
